lib/broadway/batcher.ex

```diff
@@ -47,10 +48,26 @@ defmodule Broadway.Batcher do
 
   @impl true
   def handle_events(events, _from, state) do
+    start_time = System.monotonic_time()
+    emit_start_event(state.name, start_time, events)
     batches = handle_events_per_batch_key(events, [], state)
+    emit_stop_event(state.name, start_time, batches)
```

Comment:
I'm having 2nd thoughts about `batches`. It's shape is:

```elixir
[{%Message{}, %Message{}, ...}, %BatchInfo{}]
# or:
[]
```

by exposing this as is in telemetry event, are we coupling to internal implementation that we may want to change later? Should we just emit messages without batch info (that's what we do in `broadway.processor.start`)? Or maybe expose these are `messages: [Message.t], batch_info: BatchInfo.t | nil`?

Btw, under what circumstances we have `batches == []` here, is it when we haven't filled up the batch size/batch timeout and so the current run is a no-op?

Reply:
The issue is that it can be multiple batches. My suggestion is to not emit any batchers for now. Then we add later if necessary.

Reply:
ok, I'll remove that, thanks. This way the only metadata we emit is `:name`.

Reply:
just to clarify, we still emit broadway.batcher.start,stop, right? or you'd rather skip these events altogether for now?

Reply:
Emit the events but no batchers in metadata.